tests/integration/utils.py

```diff
@@ -196,6 +196,30 @@ def make_list_of_packages_hashable(data):
     return sorted([[i["name"], i["type"], i["version"]] for i in data])
 
 
+def sort_pkgs_and_deps_in_place(packages=None, dependencies=None):
+    """
+    Sorts lists of packages and dependencies in place.
+
+    Sorting order: name -> version -> type
+
+    :param list packages: the list of packages
+    :param list dependencies: the list of dependencies
+    """
+
+    def sort_helper(elements):
+        elements.sort(key=lambda x: (x["name"], x["version"], x["type"]))
+
+    if packages is None:
+        packages = {}
+    if dependencies is None:
+        dependencies = {}
+
+    sort_helper(dependencies)
+    sort_helper(packages)
+    for package in packages:
+        sort_helper(package["dependencies"])
```

Comment:
Dicts cannot be sorted, so if you leave packages or dependencies as None, this will fail. Maybe do something like this?
```suggestion
    if packages:
        sort_helper(packages)
        for package in packages:
            sort_helper(package["dependencies"])
    if dependencies:
        sort_helper(dependencies)
```

Reply:
Yes, It was a mistake. I should use `[]`